var/spack/repos/builtin/packages/singularity/package.py

```diff
@@ -39,7 +40,7 @@ class Singularity(MakefilePackage):
     depends_on('shadow', type='run', when='@3.3:')
     depends_on('cryptsetup', type=('build', 'run'), when='@3.4:')
 
-    patch('singularity_v3.4.0_remove_root_check.patch', level=0, when='@3.4.0')
+    patch('singularity_v3.4.1_remove_root_check.patch', level=0, when='@3.4.1')
```

Comment:
The patch definitely needs to be run for 3.4.0, else it will fail.  And your believe upstream made changes such that patching may not be necessary for the new release.

Reply:
>> Also, you've indicated a new patch filename, but that file does not exist in this PR, so I think this would be a non-op.

could you explain a bit more on that pls

Reply:
You changed the name of the patch filename in the `package.py`, but you didn't add a new patch or rename the existing patch that resides in the same directory.

Reply:
oooh got it.

Reply:
reverting both changes, leaving just the addition

Reply:
Sounds good.  For the record, the mk file that the patch refers to [DID get updated](https://github.com/sylabs/singularity/blame/master/mlocal/frags/build_runtime_suid.mk) in the upstream project, so the patch should *only* be needed for 3.4.0 (as you now have it) and in fact would probably error out if you tried to apply it to 3.4.1.